assignments/callbacks.js

```diff
@@ -1,50 +1,138 @@
 // Create a higher order function and invoke the callback function to test your work. You have been provided an example of a problem and a solution to see how this works with our items array.  Study both the problem and the solution to figure out the rest of the problems.
 
 const items = ['Pencil', 'Notebook', 'yo-yo', 'Gum'];
+// const movies = ['ZombieLand', `Isn't it romantic`, 'nightmare on elm street', 'saw']
+
 
-/* 
 
   //Given this problem: 
   
-  function firstItem(arr, cb) {
-    // firstItem passes the first item of the given array to the callback function.
-  }
-
+  // function firstItem(array, callback) {
+  
+  // }
   // Potential Solution:
 
   // Higher order function using "cb" as the call back
-  function firstItem(arr, cb) {
-    return cb(arr[0]);
+  function whichItem(array, callback) {
+  console.log(callback(array));
   }
+ 
+  const firstCB = function (array) {
+    return array[0]; 
+    
+  }
+
+  // const secondCB = function (array) {
+  //   return array[1];
+  // }
+
+
+ whichItem(items, firstCB);
+
+  
+  // // Function invocation 
+  // firstItem(items, function(first) {
+    
+  // });
 
-  // Function invocation 
-  firstItem(items, function(first) {
-    console.log(first)
-  });
 
-*/
 
 
 function getLength(arr, cb) {
   // getLength passes the length of the array into the callback.
+  console.log(cb(arr));
+}
+
+const lengthCB = function (arr) {
+  return arr.length;
 }
 
+
+getLength(items, lengthCB);
+
+
+
+
+
+
 function last(arr, cb) {
   // last passes the last item of the array into the callback.
+  console.log(cb(arr));
+}
+
+const lastCB = function (arr) {
+  return arr[3];
+}
+
+last(items, lastCB);
+
+
+
+
+
+
+
+
+
+function sumNums(num1, num2, callback) {
+  return callback(num1, num2);
+  // sumNums adds two numbers (num1, num2) and passes the result to the callback.  
 }
 
-function sumNums(x, y, cb) {
-  // sumNums adds two numbers (x, y) and passes the result to the callback.
+const add = function (num1, num2) {
+  return num1 + num2;
 }
 
-function multiplyNums(x, y, cb) {
+console.log(sumNums(2, 4, add))
+
+
+function multiplyNums(num1, num2, cb) {
   // multiplyNums multiplies two numbers and passes the result to the callback.
+  return cb(num1, num2);
+}
+
+const multiply = function (num1,num2) {
+  return num1 * num2;
 }
+ 
+console.log(multiplyNums(2, 4, multiply))
+
 
-function contains(item, list, cb) {
+
+
+
+function contains(item, array, callback) {
+  return callback(item, array);
```

Comment:
Similar to the math functions above, you will want to do your logic for finding the item in the array inside of the contains function, and pass true or false into your callback